CenterForOpenScience/osf.io
# Django backward compat touch up

## Purpose

Improvements to the Django update PR
 
## Changes

- fix add license to set the licenses instead
- use ensure_str to be explicit and move it out of box to field utils 

## QA Notes

There's a few small migrations that run here, check that addon settings are unaffected.

Make sure you can:
- Add/update/remove licenses for collections/preprint/registration providers
- GDPR delete or conventionally disable a user and not effect node addons by shared projects.

## Documentation

No new docs

## Side Effects

None that I know of.

## Ticket

N/A
